marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -278,6 +282,29 @@ handleActionWithoutNavigationGuard =
         ( handleAction
         )
 
+carryMetadataAction ::
+  forall m.
+  MonadAff m =>
+  MonadAsk Env m =>
+  ME.MetadataAction ->
+  HalogenM State Action ChildSlots Void m Unit
+carryMetadataAction action = do
+  modifying (_contractMetadata) case action of
+    SetContractName name -> set _contractName name
+    SetContractType typeName -> set _contractType typeName
+    SetContractDescription description -> set _contractDescription description
+    SetRoleDescription tokenName description -> over _roleDescriptions $ Map.insert tokenName description
+    DeleteRoleDescription tokenName -> over _roleDescriptions $ Map.delete tokenName
+    SetSlotParameterDescription slotParam description -> over _slotParameterDescriptions $ Map.insert slotParam description
+    DeleteSlotParameterDescription slotParam -> over _slotParameterDescriptions $ Map.delete slotParam
+    SetValueParameterDescription valueParam description -> over _valueParameterDescriptions $ Map.insert valueParam description
+    DeleteValueParameterDescription valueParam -> over _valueParameterDescriptions $ Map.delete valueParam
+    SetChoiceDescription choiceName description -> over _choiceDescriptions $ Map.insert choiceName description
+    DeleteChoiceDescription choiceName -> over _choiceDescriptions $ Map.delete choiceName
+  metadata <- use _contractMetadata
+  assign (_hasUnsavedChanges) true
+  liftEffect $ SessionStorage.setItem metadataLocalStorageKey (encodeJSON (metadata :: MetaData))
```

Comment:
When we need to store a value of the State in the SessionStorage, instead of manually setting it, we should rely on the `Session` functionality that was introduced in [this PR](https://github.com/input-output-hk/plutus/pull/2624/files#diff-9194930343baeee1869d85e1018402dfafc9c3959e277e111c2109dfc7c6f23aR318).

This will clear up the code as you don't have to manually save and load from the session storage, the HOF `withSessionStorage` handles that for you.